apiv2/internal/server/utils.go

```diff
@@ -39,46 +44,38 @@ func toInvMetadata(apiMetadata []*commonv1.MetadataItem) (string, error) {
 	return invMetadata, nil
 }
 
-// SafeIntToUint32 converts an int to uint32 safely.
-func SafeIntToUint32(n int) (uint32, error) {
-	if n < 0 {
-		return 0, errors.New("cannot convert a negative int to uint32")
-	}
-	if n > math.MaxUint32 {
-		return 0, errors.New("int exceeds uint32 max limit")
-	}
-	return uint32(n), nil
-}
-
 // SafeInt32ToUint32 converts an int32 to uint32 safely.
 func SafeInt32ToUint32(n int32) (uint32, error) {
 	if n < 0 {
-		return 0, errors.New("cannot convert a negative int32 to uint32")
+		return 0, errors.Errorfc(codes.InvalidArgument, "cannot convert a negative int32 to uint32")
 	}
 	res := int(n)
 	if res > math.MaxUint32 && int32(res) != n { //nolint:gosec // no risk of overflow
-		return 0, errors.New("int exceeds uint32 max limit")
+		return 0, errors.Errorfc(codes.InvalidArgument, "int exceeds uint32 max limit")
 	}
 	return uint32(n), nil
 }
 
-// SafeIntToInt32 converts an int to int32 safely.
-func SafeIntToInt32(n int) (int32, error) {
-	if n < 0 {
-		return 0, errors.New("cannot convert a negative int to uint32")
-	}
-	if n > math.MaxInt32 {
-		return 0, errors.New("int exceeds uint32 max limit")
+// SafeUint32ToUint64 converts an uint32 to uint64 safely.
+func SafeUint32ToUint64(n uint32) (uint64, error) {
+	res := uint64(n)
+	if uint32(res) != n { //nolint:gocritic,gosec // no risk of overflow
+		return 0, errors.Errorfc(codes.InvalidArgument, "uint32 wrongly converted to uint64")
 	}
-	return int32(n), nil
+	return res, nil
+}
+
+// TruncateUint64ToUint32 truncates the lower bits of a uint64 to fit into a uint32.
+func TruncateUint64ToUint32(value uint64) uint32 {
+	return uint32(value & math.MaxUint32) //nolint:gosec // Mask the lower 32 bits.
```

Comment:
can we add a warning log if we go above Uint32 limit? Also, let's track on Jira the need to create uint64/string fields to be more future proof.

Reply:
Done.